spec.bs

```diff
@@ -515,14 +515,15 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
   1. Set |auctionConfig|'s [=auction config/per buyer signals=] to
     |config|["{{AuctionAdConfig/perBuyerSignals}}"].
   1. Increment |auctionConfig|'s [=auction config/pending promise count=].
-  1. [=Upon fulfillment=] of |auctionConfig|'s [=auction config/per buyer signals=] with |result|:
-    1. Execute [=validate and convert per buyer signals=] with |auctionConfig| and |result|.
-    1. If the previous step [=exception/throws=] an exception |e|:
-      1. Set |auctionConfig|'s [=auction config/per buyer signals=] to failure.
+  1. Let |resolvedAndTypeChecked| be the promise representing performing the following steps [=upon fulfillment=] of |auctionConfig|'s [=auction config/per buyer signals=] with |result|:
+    1. Set |auctionConfig|'s [=auction config/per buyer signals=] to a new [=ordered map=] whose [=map/keys=] are [=origins=] and whose [=map/values=] are [=strings=].
+    1. [=map/For each=] |key| → |value| of |result|:
+      1. Let |buyer| be the result of [=parsing an origin=] with |key|. If |buyer| is failure, throw a {{TypeError}}.
+      1. Let |signalsString| be the result of [=serializing a JavaScript value to a JSON string=], given |value|.
+      1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer signals=][|buyer|] to |signalsString|.
+    1. If the previous step did not [=exception/throw=] an exception:

```

Comment:
```suggestion
    1. If the previous steps did not [=exception/throw=] an exception:
```

You could actually remove this 'if' entirely, since if they did throw an exception that would skip any following steps, but the clarity from including it is also useful.